TotalCrossSDK/src/main/java/totalcross/xml/XmlReader.java

```diff
@@ -418,14 +419,14 @@ public final void foundEndOfInput(int count) {
    * @since TotalCross 1.27
    */
   @Override
-  protected void foundDeclaration(byte[] input, int offset, int count) {
+  protected void foundDeclaration(char[] input, int offset, int count) {
     flushAttribute();
     cntHandler.startElement(tagNameHashId, attList);
     attList.clear();
 
-    if (count > 7 && new String(totalcross.sys.Convert.charConverter.bytes2chars(input, offset, 7)).equals("[CDATA[")) {
+    if (count > 7 && new String(Arrays.copyOfRange(input, offset, 7)).equals("[CDATA[")) {
```

Comment:
Does this work outside the simulator?

Reply:
It should, I checked on Arrays4D and it has the copyOfRange, so it should work outside the simulator, couldn't test it properly because of some errors I was getting that are not related to this PR